dev/DynamicDependency/PackageGraphNode.cpp

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+
+#include "PackageGraphNode.h"
+
+volatile MDD_PACKAGEDEPENDENCY_CONTEXT MddCore::PackageGraphNode::s_lastContext{};
+
+MddCore::PackageGraphNode::PackageGraphNode(
+    _In_ PCWSTR packageFullName,
+    INT32 rank,
+    _In_ PCWSTR packageDependencyId) :
+    m_rank(rank),
+    m_id(packageDependencyId)
+{
+    THROW_IF_WIN32_ERROR(OpenPackageInfoByFullName(packageFullName, 0, &m_packageInfoReference));
+    m_packageInfo = std::move(MddCore::PackageInfo::FromPackageInfoReference(m_packageInfoReference.get()));
```

Comment:
Temporaries returned by methods are always rvalues so `x = method_that_returns_x()` invokes the right thing. No explicit `move` required.